src/App.tsx

```diff
@@ -1,24 +1,267 @@
+import { Footer } from './components/Footer';
 
-const USER_ID = 0;
+const USER_ID = 6962;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [query, setQuery] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [filterType, setFilterType] = useState(FilterTypes.ALL);
+  const [loading, setLoading] = useState(false);
+  const [loadingTodoId, setLoadingTodoId] = useState<number[]>([0]);
+
+  const closeError = () => setTimeout(() => setErrorText(''), 3000);
+  const handleCloseButton = () => setErrorText('');
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const fetchTodos = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorText('Unable to load todos');
+      closeError();
+    }
+  }, []);
+
+  useEffect(() => {
+    fetchTodos();
+  }, [fetchTodos]);
+
+  const addTodo = useCallback(async (title: string) => {
+    try {
+      setLoading(true);
+
+      if (!title.length) {
+        setErrorText("Title can't be empty");
+        closeError();
+
+        return;
+      }
+
+      const newTodo = {
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      setIsInputDisabled(true);
+      setTempTodo({ ...newTodo, id: 0 });
+
+      const createdTodo = await postTodo(newTodo);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch (error) {
+      setErrorText('Unable to add todo');
+      closeError();
+    } finally {
+      setTempTodo(null);
+      setIsInputDisabled(false);
+      setLoading(false);
+    }
+  }, []);
+
+  const removeTodo = useCallback(async (id: number) => {
+    try {
+      setLoadingTodoId(state => [...state, id]);
+      await deleteTodo(id);
+      setTodos(todos.filter(todo => todo.id !== id));
+    } catch {
+      setErrorText('Unable to delete a todo');
+      closeError();
+    } finally {
+      setLoadingTodoId([]);
+    }
+  }, [todos]);
+
+  const patchTodo = useCallback(
+    async (id: number, data: string | boolean) => {
+      setLoadingTodoId(state => [...state, id]);
+      try {
+        if (typeof data === 'boolean') {
+          await updateTodo(id, { completed: data });
+        } else {
+          await updateTodo(id, { title: data });
+        }
+
+        setTodos(todos.map(todo => {
+          if (todo.id === id) {
+            if (typeof data === 'boolean') {
+              return { ...todo, completed: data };
+            }
+
+            return { ...todo, title: data };
+          }
+
+          return todo;
+        }));
+      } catch (error) {
+        if (typeof data === 'string' && data.length === 0) {
+          setErrorText('');
```

Comment:
This function should be unique to make possible to reuse it, so better to create wrapper for title submit function and handle it there